### ISSUE TITLE
[MNG-4972] NPE in DefaultConverterLookup.findConverterForType()

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#7117]
**[Ondra Žižka](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=pekarna)** opened **[MNG-4972](https://issues.apache.org/jira/browse/MNG-4972?redirect=false)** and commented

$ mvn --version
Apache Maven 2.2.1 (r801777; 2009-08-06 21:16:01+0200)
Java version: 1.6.0_22
Java home: /usr/lib/jvm/java-6-sun-1.6.0.22/jre
Default locale: en_US, platform encoding: ANSI_X3.4-1968
OS name: "linux" version: "2.6.24-28-server" arch: "i386" Family: "unix"

Steps to reproduce:

1) Check out Hibernate Core 3.3.2
2) Download the attached settings-hbn.xml and change the \<localRepository> to some empty dir
3) cd src/core
4) mvn -s settings-hbn.xml clean install  (no profile)
5) When I run this for the first time (empty repo), I've got the exception below.

...
Downloading: http://uk.maven.org/maven2/commons-httpclient/commons-httpclient/2.0.2/commons-httpclient-2.0.2.jar
Downloading: http://uk.maven.org/maven2/jdom/jdom/1.0/jdom-1.0.jar
Downloading: http://uk.maven.org/maven2/org/codehaus/plexus/plexus-utils/1.1/plexus-utils-1.1.jar
Downloading: http://uk.maven.org/maven2/de/zeigermann/xml/xml-im-exporter/1.1/xml-im-exporter-1.1.jar
Downloading: http://uk.maven.org/maven2/commons-logging/commons-logging/1.0.4/commons-logging-1.0.4.jar

Downloading: http://uk.maven.org/maven2/slide/slide-webdavlib/2.1/slide-webdavlib-2.1.jar
Downloading: http://uk.maven.org/maven2/org/apache/maven/wagon/wagon-webdav/1.0-beta-2/wagon-webdav-1.0-beta-2.jar

[INFO] ------------------------------------------------------------------------
[INFO] Building Hibernate Core Parent POM
[INFO]    task-segment: [clean, install]
[INFO] ------------------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[ERROR] FATAL ERROR
[INFO] ------------------------------------------------------------------------
[INFO] null
[INFO] ------------------------------------------------------------------------
[INFO] Trace
java.lang.NullPointerException
at org.codehaus.plexus.component.configurator.converters.lookup.DefaultConverterLookup.findConverterForType(DefaultConverterLookup.java:115)
at org.codehaus.plexus.component.configurator.converters.lookup.DefaultConverterLookup.lookupConverterForType(DefaultConverterLookup.java:92)
at org.codehaus.plexus.component.configurator.converters.composite.CollectionConverter.fromConfiguration(CollectionConverter.java:175)
at org.codehaus.plexus.component.configurator.converters.ComponentValueSetter.configure(ComponentValueSetter.java:247)
at org.codehaus.plexus.component.configurator.converters.composite.ObjectWithFieldsConverter.processConfiguration(ObjectWithFieldsConverter.java:137)
at org.codehaus.plexus.component.configurator.BasicComponentConfigurator.configureComponent(BasicComponentConfigurator.java:56)
at org.codehaus.plexus.component.configurator.AbstractComponentConfigurator.configureComponent(AbstractComponentConfigurator.java:54)
at org.codehaus.plexus.component.configurator.AbstractComponentConfigurator.configureComponent(AbstractComponentConfigurator.java:47)
at org.codehaus.plexus.personality.plexus.lifecycle.phase.AutoConfigurePhase.execute(AutoConfigurePhase.java:39)
at org.codehaus.plexus.lifecycle.AbstractLifecycleHandler.start(AbstractLifecycleHandler.java:101)
at org.codehaus.plexus.component.manager.AbstractComponentManager.startComponentLifecycle(AbstractComponentManager.java:105)
at org.codehaus.plexus.component.manager.AbstractComponentManager.createComponentInstance(AbstractComponentManager.java:95)
at org.codehaus.plexus.component.manager.ClassicSingletonComponentManager.getComponent(ClassicSingletonComponentManager.java:92)
at org.codehaus.plexus.DefaultPlexusContainer.lookup(DefaultPlexusContainer.java:331)
at org.codehaus.plexus.DefaultPlexusContainer.lookup(DefaultPlexusContainer.java:440)
at org.apache.maven.execution.MavenSession.lookup(MavenSession.java:138)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.findMappingsForLifecycle(DefaultLifecycleExecutor.java:1358)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.bindLifecycleForPackaging(DefaultLifecycleExecutor.java:1292)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.constructLifecycleMappings(DefaultLifecycleExecutor.java:1275)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoal(DefaultLifecycleExecutor.java:534)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoalAndHandleFailures(DefaultLifecycleExecutor.java:387)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeTaskSegments(DefaultLifecycleExecutor.java:348)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.execute(DefaultLifecycleExecutor.java:180)
at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:328)
at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:138)
at org.apache.maven.cli.MavenCli.main(MavenCli.java:362)
at org.apache.maven.cli.compat.CompatibleMain.main(CompatibleMain.java:60)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
at java.lang.reflect.Method.invoke(Method.java:597)
at org.codehaus.classworlds.Launcher.launchEnhanced(Launcher.java:315)
at org.codehaus.classworlds.Launcher.launch(Launcher.java:255)
at org.codehaus.classworlds.Launcher.mainWithExitCode(Launcher.java:430)
at org.codehaus.classworlds.Launcher.main(Launcher.java:375)

---

**Affects:** 2.2.1

**Attachments:**
- [settings-hbn.xml](https://issues.apache.org/jira/secure/attachment/12713148/settings-hbn.xml) (_2.52 kB_)

**Issue Links:**
- [MNG-4947](https://issues.apache.org/jira/browse/MNG-4947) NPE in at org.codehaus.plexus.component.configurator.converters.lookup.DefaultConverterLookup.findConverterForType()
 (_**"duplicates"**_)

**Comment by jira-importer** [comment]
**[Ondra Žižka](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=pekarna)** commented

Attaching settings.xml.
Also change the path to the JDK 1.6.
Build with JDK 1.5 !!

**Comment by jira-importer** [comment]
**[Benjamin Bentmann](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=bentmann)** commented

Out of curiosity, what changed since MNG-4947 that makes it reproducible for you now, since I still don't see the failure.

In general, your steps to repro are unclear/incomplete:

> 1) Check out Hibernate Core 3.3.2
>    I assume this means https://anonsvn.jboss.org/repos/hibernate/core/tags/hibernate-3.3.2.GA_CP03
> 2) cd src/core
>    There is no directory `src/core` in the above mentioned Hibernate tag. While there is a `core` directory, this builds a project named "Hibernate Core" and not "Hibernate Core Parent POM" as shown in your log. so I assume you meant `cd hibernate-3.3.2.GA_CP03`, i.e. build the aggregator POM located in the checkout directory
> 3) mvn -s settings-hbn.xml clean install (no profile)
>    Using the provided settings and building without profiles fails due to an unresolvable artifact (org.jboss.maven.plugins:maven-test-ext-plugin:pom:1.1.0), so I assume you meant to enable `jboss-repos` profile.
> 4) When I run this for the first time (empty repo), I've got the exception below.
>    Works here:

```
M:\downloads\hibernate-3.3.2.GA_CP03>mvn -V -s settings-hbn.xml clean install -e -P jboss-repos
+ Error stacktraces are turned on.
Apache Maven 2.2.1 (r801777; 2009-08-06 21:16:01+0200)
Java version: 1.6.0_21
Java home: D:\java\jdk-1.6\jre
Default locale: de_DE, platform encoding: Cp1252
OS name: "windows 7" version: "6.1" arch: "x86" Family: "windows"
[INFO] Scanning for projects...
[INFO] Reactor build order:
[INFO]   Hibernate Core Parent POM
[INFO]   Hibernate Core
[INFO]   Hibernate Testing
[INFO]   Hibernate Ehcache Integration
[INFO]   Hibernate JBossCache2.x Integration
[INFO]   Hibernate OSCache Integration
[INFO]   Hibernate SwarmCache Integration
[INFO]   Hibernate C3P0 ConnectionProvider
[INFO]   Hibernate Proxool ConnectionProvider
[INFO]   Hibernate JDBC4-JdbcSupport Testing
[INFO]   Hibernate JDBC3-JdbcSupport Testing
[INFO]   Hibernate JMX Module
[INFO]   Hibernate Testsuite
[INFO]   Hibernate Tutorials
[INFO]   Hibernate Example
[INFO]   Hibernate Web Tutorial
[INFO]   Hibernate Distribution
[INFO]   Hibernate Core Aggregator
...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] ------------------------------------------------------------------------
[INFO] Hibernate Core Parent POM ............................. SUCCESS [6:16.831s]
[INFO] Hibernate Core ........................................ SUCCESS [4:03.493s]
[INFO] Hibernate Testing ..................................... SUCCESS [0.620s]
[INFO] Hibernate Ehcache Integration ......................... SUCCESS [5.456s]
[INFO] Hibernate JBossCache2.x Integration ................... SUCCESS [31.500s]
[INFO] Hibernate OSCache Integration ......................... SUCCESS [5.105s]
[INFO] Hibernate SwarmCache Integration ...................... SUCCESS [12.117s]
[INFO] Hibernate C3P0 ConnectionProvider ..................... SUCCESS [5.456s]
[INFO] Hibernate Proxool ConnectionProvider .................. SUCCESS [8.453s]
[INFO] Hibernate JDBC4-JdbcSupport Testing ................... SUCCESS [0.630s]
[INFO] Hibernate JDBC3-JdbcSupport Testing ................... SUCCESS [0.690s]
[INFO] Hibernate JMX Module .................................. SUCCESS [1.190s]
[INFO] Hibernate Testsuite ................................... SUCCESS [1:52.561s]
[INFO] Hibernate Tutorials ................................... SUCCESS [0.020s]
[INFO] Hibernate Example ..................................... SUCCESS [0.410s]
[INFO] Hibernate Web Tutorial ................................ SUCCESS [1:00.109s]
[INFO] Hibernate Distribution ................................ SUCCESS [1:33.220s]
[INFO] Hibernate Core Aggregator ............................. SUCCESS [10.800s]
[INFO] ------------------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESSFUL
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 16 minutes 40 seconds
[INFO] Finished at: Thu Jan 13 13:47:11 CET 2011
[INFO] Final Memory: 38M/176M
[INFO] ------------------------------------------------------------------------
```

Last but not least, there is work planned for the Plexus container used by Maven 2.x. so if this issue doesn't show up in Maven 3.0, it doesn't have any relevance.

**Comment by jira-importer** [comment]
**[Ondra Žižka](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=pekarna)** commented

I, sorry for duplicate, I couldn't find so I thought I didn't file.

1) Some of your assumptions are correct. It's ..._CR3,
2) The src/core came from my build structure. So, I meant the root pom, Hibernate Agregator
3) I intentionally wrote (no profile) because that's what I done when I got the exception.
   I know that it's not the way to build the project, but it's a record how I got the bug which bugs me now and then in correctly configured builds.
4) I also get this bug rarely. However when I get it, it spoils quite a long chain of runs in continuous integration.
5) Anyway, the bug does not appear often, so if it's going to be buried with upcoming Maven 3, so be it.

Thanks for trying.
